docs/spark/how-to-guides/debug.md

```diff
@@ -30,13 +30,40 @@ When you run the command, you see the following output:
 ***********************************************************************
 ```
 
-In this debug mode, `DotnetRunner` does not launch the .NET application, but it waits for it to connect. Leave this command prompt window open.
+In this debug mode, `DotnetRunner` does not launch the .NET application, but waits for it to connect. Leave this command prompt window open.
 
-Now you can run your .NET application with any debugger to debug your application.
+Now you can start your .NET application with a C# debugger ([Visual Studio Debugger for Windows/macOS](https://visualstudio.microsoft.com/vs/) or [C# Debugger Extension in Visual Code](https://code.visualstudio.com/Docs/editor/debugging)) to debug your application.
 
-## Debug Scala code
+## Debugging User Defined Function (UDF)
 
-If you need to debug the Scala side code, such as `DotnetRunner` or `DotnetBackendHandler`, run the following command:
+**Note that this is currently supported only on Windows with Visual Studio Debugger.**
+
+Before running `spark-submit`, set the following environment variable:
+
+```bat
+set DOTNET_WORKER_DEBUG=1
+```
+
+Now, when you run your Spark application, a `Choose Just-In-Time Debugger` window will pop up. Choose a Visual Studio debugger.
+
+The debugger will break at the following location in [TaskRunner.cs](../src/csharp/Microsoft.Spark.Worker/TaskRunner.cs):
```

Comment:
Is this link valid ? I don't think.

Reply:
oh yes, this is relative to the Spark .NET repo, will add the absolute link, thanks!